pkg/logger/logger.go

```diff
@@ -0,0 +1,289 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package logger
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"github.com/dapr/dapr/pkg/version"
+	"github.com/dapr/dapr/utils"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// LogTypeLog is normal log type
+	LogTypeLog = "log"
+	// LogTypeRequest is Request log type
+	LogTypeRequest = "request"
+
+	// Field names that defines Dapr log schema
+	logFieldTimeStamp = "time"
+	logFieldLevel     = "level"
+	logFieldType      = "log_type"
+	logFieldScope     = "scope"
+	logFieldMessage   = "message"
+	logFieldInstance  = "instance"
+	logFieldDaprVer   = "dapr_ver"
+	logFieldDaprID    = "dapr_id"
+)
+
+// LogLevel is Dapr Logger Level type
+type LogLevel uint32
+
+const (
+	// DebugLevel has verbose message
+	DebugLevel LogLevel = iota
+	// InfoLevel is default log level
+	InfoLevel
+	// WarnLevel is for logging messages about possible issues
+	WarnLevel
+	// ErrorLevel is for logging errors
+	ErrorLevel
+	// FatalLevel is for logging fatal messages. The system shutsdown after logging the message.
+	FatalLevel
+)
+
+var daprLogLevelToLogrusLevel = map[LogLevel]logrus.Level{
```

Comment:
Not sure we need this map. It would be better to declare string types instead, and then use `ParseLevel` in logrus to get the logrus.Level value.

Example:

```go
type LogLevel string

const (
	InfoLevel LogLevel = "info"
)
```

Reply:
addressed this.